### PR TITLE
keyboard shortcut to toggle snapping

### PR DESCRIPTION
**Pull request by 3nids**
Adds a new keyboard shortcut ("S" by default) to quickly toggle snapping on and off.

### CODE REVIEW

**Comment by m-kuhn** [comment]
@3nids Force pushing on a branch shared via pull request works [QED]

File: scripts/sipdiff [hunk 1]
```diff
@@ -53,7 +53,7 @@ for file in $*; do
   ${GP}sed -i -r '/^\s*Q_(OBJECT|ENUMS|PROPERTY).*?$/d' $tempfile
 
   # Remove function definition in header
-  ${GP}sed -i -r 's/^(\s*)?(virtual |static )?(inline )?(void|bool|int|double|Q\w+)(\s+[^ ]*?\(.*?\)( const)?)\s*\{.*?\}$/\1\2\4\5;/g' $tempfile
+  ${GP}sed -i -r 's/^(\s*)?(virtual |static )?(inline )?(void|bool|int|double|Q\w+)(\*?))(\s+[^ ]*?\(.*?\)( const)?)\s*\{.*?\}$/\1\2\4\5\6;/g' $tempfile
```
Comment (3nids) [inline scripts/sipdiff]: I think there is an extra closing parenthesis here, right?

**Comment by 3nids** [comment]
thanks for the info

File: scripts/sipdiff (hunk 1 above)
Comment (m-kuhn) [inline scripts/sipdiff]: Probably, I didn't touch it ;)

Comment (3nids) [inline scripts/sipdiff]: ok, I'll push here too then

Comment (3nids) [inline scripts/sipdiff]: thanks...

**Pull request merged**